### PR TITLE
Allow omicron to configure per-link tx_eq settings

### PR DESCRIPTION
**Pull request by Nieuwejaar**

### CODE REVIEW

File: common/src/api/external/mod.rs [hunk 1]
```diff
@@ -2355,6 +2355,10 @@ pub struct SwitchPortSettingsView {
     /// Link-layer discovery protocol (LLDP) settings.
     pub link_lldp: Vec<LldpLinkConfig>,
 
+    /// TX equalization settings.  These are optional, and most links will not
+    /// need them.
+    pub tx_eq: Vec<Option<TxEqConfig>>,
```
Comment (internet-diglett) [inline common/src/api/external/mod.rs:2360]: Is this collection truly variable in length, or can we expect a fixed length of optional values?

File: common/src/api/external/mod.rs [hunk 2]
```diff
@@ -2355,6 +2355,10 @@ pub struct SwitchPortSettingsView {
     /// Link-layer discovery protocol (LLDP) settings.
     pub link_lldp: Vec<LldpLinkConfig>,
 
+    /// TX equalization settings.  These are optional, and most links will not
+    /// need them.
```
Comment (internet-diglett) [inline common/src/api/external/mod.rs:2359]: When will links need this setting? Is there an issue / document we can reference for this feature?

**Review by internet-diglett — COMMENTED** [review]
Hey @Nieuwejaar, I attached a few questions in order to get a better understand of how this feature is going to be used. Will we need to expose this to customers via the external api?

File: common/src/api/external/mod.rs (hunk 2 above)
Comment (Nieuwejaar) [reply common/src/api/external/mod.rs:2359]: This is needed for specific models of transceiver which don't autotune and don't work with the normal default settings.  So far, we've only found one of those, and that is now hardcoded in `dendrite`.  For now, this is for internal use only, by whoever is doing the initial customer install.

File: common/src/api/external/mod.rs (hunk 1 above)
Comment (Nieuwejaar) [reply common/src/api/external/mod.rs:2360]: Until we add support for breakout connectors and multiple links per port, the length will always be 1.  I implemented this as a vector, as that is how all of the other per-link settings are represented in this struct.

**Review by internet-diglett — APPROVED** [review]

**Pull request merged**